src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+let headers, tbody, rows;
+
+const init = () => {
+  headers = [...body.querySelectorAll('th')];
+  tbody = body.querySelector('tbody');
+  rows = [...tbody.querySelectorAll('tr')];
+};
+
+init();
+
+const currencyFormat = (number) =>
+  '$' + new Intl.NumberFormat('en-US').format(number);
+
+// Add active class to row when clicked
+
+rows.forEach((row) => {
+  row.addEventListener('click', () => {
+    rows.forEach((r) => r.classList.remove('active'));
+    row.classList.toggle('active');
+  });
+});
+
+let ASC = true;
+let currentHeader;
+
+// Sort table by column
+headers.forEach((header, i) => {
+  header.addEventListener('click', (e) => {
+    ASC = !ASC;
+
+    if (currentHeader !== i) {
+      ASC = true;
+    }
+    currentHeader = i;
+
+    rows.sort((a, b) => {
+      const cellA = a.querySelectorAll('td')[i].textContent;
+      const cellB = b.querySelectorAll('td')[i].textContent;
+
+      const checkNumberCellA = strToNumber(cellA);
+      const checkNumberCellB = strToNumber(cellB);
+
+      if (Number(checkNumberCellA)) {
+        return ASC
+          ? checkNumberCellA - checkNumberCellB
+          : checkNumberCellB - checkNumberCellA;
+      }
+
+      return ASC ? cellA.localeCompare(cellB) : cellB.localeCompare(cellA);
+    });
+
+    rows.forEach((row) => tbody.appendChild(row));
+  });
+});
```

Comment:
This block of code is responsible for sorting table rows when a table header is clicked. It would be more efficient to separate this logic into smaller functions, for example, a function that gets the sort direction, a function that gets the cell value, a function that compares cell values, and a function that sorts the rows. This would make the code more readable and maintainable.